File: backend/app/schemas/audit.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class AuditActorRead(BaseModel):
    id: UUID | None
    full_name: str | None
    email: str | None


class AuditEventRead(BaseModel):
    id: UUID
    occurred_at: datetime
    actor: AuditActorRead
    action_type: str
    entity_type: str
    entity_id: UUID | None
    status: str
    summary: str
    entity_label: str | None = None
    product_id: UUID | None = None
    product_release_id: UUID | None = None
    details_json: dict = Field(default_factory=dict)


class AuditEventListRead(BaseModel):
    items: list[AuditEventRead]
    total: int


class AuditIntegrityIssueRead(BaseModel):
    sequence_number: int | None = None
    event_id: UUID | None = None
    reason: str


class AuditIntegrityRead(BaseModel):
    verified: bool
    total_events: int
    verified_events: int
    latest_sequence_number: int | None = None
    issues: list[AuditIntegrityIssueRead] = Field(default_factory=list)

File: frontend/src/types/audit.ts
export interface AuditActorRead {
  id: string | null;
  full_name: string | null;
  email: string | null;
}

export interface AuditEventRead {
  id: string;
  occurred_at: string;
  actor: AuditActorRead;
  action_type: string;
  entity_type: string;
  entity_id: string | null;
  status: string;
  summary: string;
  entity_label: string | null;
  product_id: string | null;
  product_release_id: string | null;
  details_json: Record<string, unknown>;
}

export interface AuditEventListRead {
  items: AuditEventRead[];
  total: number;
}

export interface AuditIntegrityIssueRead {
  sequence_number: number | null;
  event_id: string | null;
  reason: string;
}

export interface AuditIntegrityRead {
  verified: boolean;
  total_events: number;
  verified_events: number;
  latest_sequence_number: number | null;
  issues: AuditIntegrityIssueRead[];
}

export interface AuditEventListParams {
  entity_id?: string;
  product_id?: string;
  product_release_id?: string;
  actor_user_id?: string;
  action_type?: string;
  action_prefix?: string;
  entity_type?: string;
  limit?: number;
}
